pkg/controller/backup_status_updater.go

```diff
@@ -75,12 +82,18 @@ func NewRealBackupConditionUpdater(
 func (u *realBackupConditionUpdater) Update(backup *v1alpha1.Backup, condition *v1alpha1.BackupCondition, newStatus *BackupUpdateStatus) error {
 	ns := backup.GetNamespace()
 	backupName := backup.GetName()
-	var isUpdate bool
 	// try best effort to guarantee backup is updated.
 	err := retry.OnError(retry.DefaultRetry, func(e error) bool { return e != nil }, func() error {
-		updateBackupStatus(&backup.Status, newStatus)
-		isUpdate = v1alpha1.UpdateBackupCondition(&backup.Status, condition)
-		if isUpdate {
+		var isStatusUpdate, isConditionUpdate, isLogSubcommandStatusUpdate bool
+		// log backup status update should be handled specially before backup is delete.
+		// it needs update both subcommand status and whole backup status.
+		if backup.Spec.Mode == v1alpha1.BackupModeLog && backup.DeletionTimestamp == nil {
```

Comment:
For BackupModeLog, branch if and else will be executed at different situations. For BackupModeSnapshot, only branch else will be executed. Since these two Modes are very different, can we organize status update logic like this?
if backup.Spec.Mode == v1alpha1.BackupModeLog{
       // do sth
}else{
      // do another thing
}
Make it easy to understand.